plugins/BEdita/Core/src/Model/Table/ObjectTypesTable.php

```diff
@@ -348,6 +348,24 @@ public function findAll(Query $query, array $options)
         return $query->contain(['LeftRelations', 'RightRelations']);
     }
 
+    /**
+     * Find object types having a parent by `name` or `id`
+     *
+     * @param \Cake\ORM\Query $query Query object.
+     * @param array $options Additional options. The first element containing `id` or `name` is required.
+     * @return \Cake\ORM\Query
+     * @throws \BEdita\Core\Exception\BadFilterException When missing required parameters.
+     */
+    public function findParent(Query $query, array $options)
+    {
+        if (empty($options[0])) {
+            throw new BadFilterException(__d('bedita', 'Missing required parameter "{0}"', 'parent'));
+        }
+        $parentId = $this->get($options[0])->get('id');
```

Comment:
Loading the whole object type is not necessary if the option is already numeric:
```php
$parentId = $options[0];
if (!is_numeric($parentId)) {
    $parentId = $this->get($parentId)->id;
}
```

Or, probably even better, we could extract the part of `ObjectTypes::get()` that builds the name-to-id map and use it here.

Reply:
Ok, did only the numeric check